src/Features/Core/Portable/ChangeSignature/Parameter.cs

```diff
@@ -51,21 +51,25 @@ internal sealed class AddedParameter : Parameter
 
             IsRequired = isRequired;
             DefaultValue = defaultValue;
-            IsCallsiteTodo = isCallsiteTodo;
-            IsCallsiteOmitted = isCallsiteOmitted;
-            UseNamedArguments = useNamedArguments;
+            CallSiteKind = callSiteKind;
 
-            if (IsCallsiteTodo)
-            {
-                CallSiteValue = FeaturesResources.ChangeSignature_NewParameterIntroduceTODOVariable;
-            }
-            else if (isCallsiteOmitted)
-            {
-                CallSiteValue = FeaturesResources.ChangeSignature_NewParameterOmitValue;
-            }
-            else
+            switch (CallSiteKind)
             {
-                CallSiteValue = callSiteValue;
+                case CallSiteKind.Value:
+                case CallSiteKind.ValueWithName:
+                    CallSiteValue = callSiteValue;
+                    break;
+                case CallSiteKind.Todo:
+                    CallSiteValue = FeaturesResources.ChangeSignature_NewParameterIntroduceTODOVariable;
+                    break;
+                case CallSiteKind.Omitted:
+                    CallSiteValue = FeaturesResources.ChangeSignature_NewParameterOmitValue;
+                    break;
+                case CallSiteKind.Inferred:
+                    CallSiteValue = FeaturesResources.ChangeSignature_NewParameterInferValue;
```

Comment:
confused about what's hpapneing here.  why are we assigning resource strings into this value?

Reply:
oh, is this for teh UI?  if so, doc this (And the property below).

Reply:
Updated.